src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -179,6 +179,36 @@ static void uct_rc_mlx5_iface_progress_enable(uct_iface_h tl_iface, unsigned fla
                                       iface->super.progress, flags);
 }
 
+static ucs_status_t uct_rc_mlx5_iface_create_qp(uct_ib_iface_t *iface,
+                                                uct_ib_qp_attr_t *attr,
+                                                struct ibv_qp **qp_p)
+{
+#if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE
```

Comment:
let's always use mlx5dv_create_qp for rc_mlx5, if mlx5dv_create_qp  exists.
then, if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE, pass this flag.
(i'm thinking about other reasons we might want to use mlx5dv_create_qp )